code/modules/wiremod/shell/module.dm

```diff
@@ -90,66 +93,122 @@
+		if(potential_module.name == module_to_select.value || potential_module == module_to_select.value)
 			module = potential_module
 	if(COMPONENT_TRIGGERED_BY(toggle_suit, port))
 		INVOKE_ASYNC(attached_module.mod, /obj/item/mod/control.proc/toggle_activate, attached_module.mod.wearer)
+	if(COMPONENT_TRIGGERED_BY(toggle_deploy, port))
+		INVOKE_ASYNC(attached_module.mod, /obj/item/mod/control.proc/quick_deploy, attached_module.mod.wearer)
 	if(attached_module.mod.active && module && COMPONENT_TRIGGERED_BY(select_module, port))
 		INVOKE_ASYNC(module, /obj/item/mod/module.proc/on_select)
 
 /obj/item/circuit_component/mod_adapter_core/proc/on_move(atom/movable/source, atom/old_loc, dir, forced)
 	SIGNAL_HANDLER
 	if(istype(source.loc, /obj/item/mod/control))
-		RegisterSignal(source.loc, COMSIG_MOD_MODULE_SELECTED, .proc/on_module_select)
-		RegisterSignal(source.loc, COMSIG_ITEM_EQUIPPED, .proc/equip_check)
-		equip_check()
+		var/obj/item/mod/control/mod = source.loc
+		RegisterSignal(mod, COMSIG_MOD_MODULE_SELECTED, .proc/on_module_select)
+		RegisterSignal(mod, COMSIG_MOD_PART_TOGGLED, .proc/on_mod_part_toggled)
+		RegisterSignal(mod, COMSIG_MOD_TOGGLED, .proc/on_mod_toggled)
+		RegisterSignal(mod, COMSIG_MOD_MODULE_ADDED, .proc/on_module_changed)
+		RegisterSignal(mod, COMSIG_MOD_MODULE_REMOVED, .proc/on_module_changed)
+		RegisterSignal(mod, COMSIG_ITEM_EQUIPPED, .proc/equip_check)
+		wearer.set_output(mod.wearer)
+		module_to_select.possible_options = mod.modules
+		if (module_to_select.possible_options.len)
+			module_to_select.set_value(module_to_select.possible_options[1])
 	else if(istype(old_loc, /obj/item/mod/control))
 		UnregisterSignal(old_loc, list(COMSIG_MOD_MODULE_SELECTED, COMSIG_ITEM_EQUIPPED))
+		UnregisterSignal(old_loc, COMSIG_MOD_PART_TOGGLED)
+		UnregisterSignal(old_loc, COMSIG_MOD_TOGGLED)
+		UnregisterSignal(old_loc, COMSIG_MOD_MODULE_ADDED)
+		UnregisterSignal(old_loc, COMSIG_MOD_MODULE_REMOVED)
 		selected_module.set_output(null)
 		wearer.set_output(null)
+		deployed.set_output(FALSE)
+		activated.set_output(FALSE)
 
 /obj/item/circuit_component/mod_adapter_core/proc/on_module_select(datum/source, obj/item/mod/module/module)
 	SIGNAL_HANDLER
 	selected_module.set_output(module.name)
 	on_module_selected.set_output(COMPONENT_SIGNAL)
 
+/obj/item/circuit_component/mod_adapter_core/proc/on_module_changed()
+	SIGNAL_HANDLER
+	module_to_select.possible_options = attached_module.mod.modules
+	if (module_to_select.possible_options.len)
+		module_to_select.set_value(module_to_select.possible_options[1])
 
-/obj/item/circuit_component/mod_adapter_core/proc/equip_check()
+
+/obj/item/circuit_component/mod_adapter_core/proc/on_mod_part_toggled()
 	SIGNAL_HANDLER
+	var/is_deployed = TRUE
+	for(var/obj/item/part as anything in attached_module.mod.mod_parts)
+		if(part.loc == src)
+			is_deployed = FALSE
+			break
+	deployed.set_output(is_deployed)
+	on_deploy.set_output(COMPONENT_SIGNAL)
```

Comment:
honestly, it may just be a good idea to separate these out into two signals, COMSIG_MOD_PART_DEPLOYED and COMSIG_MOD_PART_UNDEPLOYED

Reply:
Should I seperate the output signals too on the component?

And should I seperate COMSIG_MOD_TOGGLED too?